sdk/log/batch.go

```diff
@@ -96,6 +96,8 @@ type BatchProcessor struct {
 
 	// stopped holds the stopped state of the BatchProcessor.
 	stopped atomic.Bool
+
+	noCmp [0]func() //nolint: unused  // This is indeed used.
```

Comment:
```suggestion
	noCmp [0]func() //nolint: unused  // This is indeed used to make this struct non-comparable.
```

Nit. To make it clear.

Reply:
I'm not in favor of that change. It makes the line too long, is verbose without adding addition information. From the context of the field type it is already implied that this is making the struct non-comparable.

This is a sentence we have used many times before, and continues to be appropriate at this point.